src/neofs_testlib/env/env.py

```diff
@@ -0,0 +1,486 @@
+import json
+import logging
+import os
+import pickle
+import random
+import socket
+import string
+import subprocess
+from dataclasses import dataclass
+from enum import Enum
+from importlib.resources import files
+from pathlib import Path
+from typing import Optional
+
+import allure
+import jinja2
+from tenacity import retry, stop_after_attempt, wait_fixed
+
+from neofs_testlib.cli import NeofsAdm, NeofsCli
+from neofs_testlib.shell import LocalShell
+from neofs_testlib.utils import wallet as wallet_utils
+
+logger = logging.getLogger("neofs.testlib.env")
+
+
+@dataclass
+class NodeWallet:
+    path: str
+    address: str
+    password: str
+
+
+class WalletType(Enum):
+    STORAGE = 1
+    ALPHABET = 2
+
+
+class NeoFSEnv:
+    def __init__(self):
+        self.domain = "localhost"
+        self.default_password = "password"
+        self.shell = LocalShell()
+        # utilities
+        self.neofs_adm_path = os.getenv("NEOFS_ADM_BIN", "./neofs-adm")
+        self.neofs_cli_path = os.getenv("NEOFS_CLI_BIN", "./neofs-cli")
+        self.neo_go_path = os.getenv("NEO_GO_BIN", "./neo-go")
+        self.neofs_ir_path = os.getenv("NEOFS_IR_BIN", "./neofs-ir")
+        self.neofs_node_path = os.getenv("NEOFS_NODE_BIN", "./neofs-node")
+        self.neofs_s3_authmate_path = os.getenv("NEOFS_S3_AUTHMATE_BIN", "./neofs-s3-authmate")
+        self.neofs_s3_gw_path = os.getenv("NEOFS_S3_GW_BIN", "./neofs-s3-gw")
+        self.neofs_rest_gw_path = os.getenv("NEOFS_REST_GW_BIN", "./neofs-rest-gw")
+        self.neofs_http_gw_path = os.getenv("NEOFS_HTTP_GW_BIN", "./neofs-http-gw")
```

Comment:
I'd better have gateways defined separately somewhere, many tests don't care about them at all. Those who care, should just run an instance as needed.

Reply:
This is not a definition of gateways, it is just the way to organize env vars in one place. If tests don't care - they won't use it, there is no requirement to specify any of these env vars. Gateways are indeed separate classes (you can see the class for s3 gateway in this PR). I can move related paths there, but it seems to be more convenient to have them in one place.